web/components/PetPostModal/DeleteButtonModal.tsx

```diff
@@ -0,0 +1,107 @@
+import { DeleteIcon } from "@chakra-ui/icons";
+import {
+  Button,
+  Flex,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  useDisclosure,
+  useToast,
+} from "@chakra-ui/react";
+import { trpc } from "../../utils/trpc";
+import { Types } from "mongoose";
+
+function DeleteButtonModal(props: {
+  onClose: () => void;
+  postId: Types.ObjectId;
+}) {
+  const { onClose, postId } = props;
+
+  const mutation = trpc.post.delete.useMutation();
+  const utils = trpc.useContext();
+  const toast = useToast();
+
+  const {
+    isOpen: isDeleteConfirmationOpen,
+    onOpen: onDeleteConfirmationOpen,
+    onClose: onDeleteConfirmationClose,
+  } = useDisclosure();
+
+  return (
+    <Flex>
+      <Button
+        h={8}
+        backgroundColor="white"
+        onClick={onDeleteConfirmationOpen}
+        _hover={{}}
+        leftIcon={<DeleteIcon marginRight="5px" color="text-secondary" />}
+      >
+        <Text textDecoration="underline" color="text-secondary">
+          Delete
+        </Text>
+      </Button>
```

Comment:
Move this back into the Post modal so that the deletion modal component only contains the modal itself